### PR TITLE
Introduce manifest entry, meta and their serializers

### PR DESCRIPTION
**Pull request by tsreaper**

### CODE REVIEW

File: flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/ValueType.java [hunk 1]
```diff
@@ -18,31 +18,31 @@
 
 package org.apache.flink.table.storage.filestore;
 
-/** Value kind of a record. */
-public enum ValueKind {
+/** Value type of a record. */
+public enum ValueType {
```
Comment (JingsongLi) [inline flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/ValueType.java]: Do not use `Type`, there was a discussion before and the conclusion was that `type` was ambiguous.

Comment (tsreaper) [reply flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/ValueType.java]: OK, let me revert this.

File: flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/stats/FieldStatsSerializeUtils.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import org.apache.flink.table.types.logical.ArrayType;
+import org.apache.flink.table.types.logical.BigIntType;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+/** Util methods for (de)serializing {@link FieldStats}. */
+public class FieldStatsSerializeUtils {
```
Comment (JingsongLi) [inline flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/stats/FieldStatsSerializeUtils.java]: Can we create a `FieldStatsArraySerializer`?

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**